tango/common/util.py

```diff
@@ -153,3 +159,16 @@ def could_be_class_name(name: str) -> bool:
 
 def _is_valid_python_name(name: str) -> bool:
     return bool(name and name[0].isalpha() and name.isalnum())
+
+
+def find_open_port() -> int:
```

Comment:
Now that you changed how server startup works, do we still need this?

Reply:
Yes, this is called in a loop until an open port is successfully claimed. https://github.com/allenai/tango/blob/15ea0ad4eda27c29df41b864e4d5b08a436c3313/tango/common/logging.py#L445-L452

Reply:
I clearly did not read this carefully enough before. I thought you just did what I did and counted up ports. But you just pass `0` for the port and let the OS handle it! Brilliant!

I just tried this with `socketserver.TCPServer`, and found that you can pass a `0` there too and it'll pick a port. `TCPServer` is the base class of `LogRecordSocketReceiver`, so I think it'll work there too. No utilities required at all.

Reply:
Oh, cool. I'll try that.

Reply:
96a283a